### DISCUSSION TITLE
Display a custom title in the app and on lock screen notifications for better identification with multiple users on one device

### DISCUSSION THREAD

**Discussion by Pescador76** [NightscoutFoundation/xDrip#2573] [Ideas]
Hello and many thanks to the developers of this app. We are parents of two children with diabetes. On my smartphone I have two versions of xdrip+ in follower mode, so I can see the values of both children at any time. The only problem is that the lockscreen notifications do not tell me which child the displayed value refers to. Also in the app there is nowhere the possibility to store a name or a recognition feature. In Nightscout, for example, there is the option to store a user-defined title, which is then clearly visible on the overview page. My only wish for the xdrip+ app would be to be able to enter a custom title to be displayed in the notifications and in the app. Thanks again to all of you.

**Comment by Navid200** [comment]
Thanks for the suggestion.

Can I ask you to read this first:  https://github.com/NightscoutFoundation/xDrip/issues/267

Then, if you think you still cannot accomplish what you need, would you let me know then please?

**Reply by Pescador76** [reply]
Yes, I read this discussion before. And this is exactly what I did: I installed two variants of xdrip and now both values are displayed. The only problem is that it is easy to confuse which display refers to which patient or which child. In one variant it says "xdrip+" and in the other "xdrip+ variant4". However, this is easy to get confused with. Entering a custom title or custom label would make it easier to distinguish the versions.

![grafik](https://user-images.githubusercontent.com/13404652/208946431-471ee330-0a53-4f0e-afb2-2f18096fa1b1.png)

To make it even easier to understand, here is a screenshot of my lockscreen. The top notification is from the dexcom app, it shows the name of the child and the current blood glucose value. Below is the xdrip app notification, it only shows the value without the name. Now if I have two xdrip+ notifiactions for two children, I don't know which one belongs to whom.

**Reply by Navid200** [reply]
I respectfully disagree.  Please have a look here: https://navid200.github.io/xDrip/docs/Variants
The variant is clearly identified and one can tell which is which.

Can you show an example that shows otherwise.
If it is clear that something is missing, we will address it.  I just would like to make sure you are aware of all the existing options.

**Reply by Pescador76** [reply]
Sorry for the misunderstanding. I know that the different variants names are displayed. But my suggestion was that instead of just seeing the app name, you make it possible to enter your own name or label. Because if I get woken up by an alarm at three in the morning, it's easier if I can see directly which child is affected. If I only see the app name, then I first have to think about which child has which app name assigned to it. Imagine that more than two children are affected. Then you quickly get confused. Therefore the suggestion that you can enter a name, just as it is possible in the Nighscout Webclient. 

Such a display would be great:
![grafik](https://user-images.githubusercontent.com/13404652/208990957-21c9f42e-1244-4968-a3a4-de1663e70926.png)

In Nighscout webclient, you can set a Custom title, which is very comfortable:
![grafik](https://user-images.githubusercontent.com/13404652/208991730-ab25960f-0594-437f-bd1f-3295d27bfe78.png)

**Reply by Navid200** [reply]
I have explained on that same page how you can get identifiable alerts by naming the alerts in each app accordingly.
Did you see that?
Because reading your post, again I cannot help wondering if you have seen that or not.

**Reply by Pescador76** [reply]
Yes, I actually considered the possibility of sound assignment as well. So currently you only have the option of using sounds, colors or other 'placeholders' to assign a patient. I still maintain that a placeholder is still an error-prone substitute for a unique designation. Quite simply, it would be optimal to be able to enter a name somewhere in xdrip that would then be displayed everywhere. It would make it easier for those who need to monitor multiple patients. 

I wrote it here especially in the forum "ideas", because I think it is a good idea, which is worth to implement and which probably does not even require a lot of effort. But if you see it differently, fine. You guys are doing a great job here.

**Reply by Navid200** [reply]
Would you please try it out?
I still don't think you are considering what is being suggested.

When you get an alert, the name of the alert is shown in the notification.  So, your summary of categorizing that as "using sounds" is not valid categorization.
It is true that you can pick a specific sound for the alert.  of course.  But, that wasn't the point.

**Reply by Pescador76** [reply]
Got it. I try that out, thank you with personalized alerts. In the event of an alarm, this is then sufficient as an identification feature. I agree with you there.
But this only affects the alarms, the normal notification in the lockscreen is still somewhat unclear.

**Reply by nightscout-ricsi** [reply]
I would like to have this labeling, too. On the home screen (of xDrip) and on the lockscreen notifications. It would be nice to change "xdrip+" and "xdrip+ variant4" to the names of the patients.

**Reply by Navid200** [reply]
Can you please show a screenshot of what you see on the lockscreen.  
What you posted before showed Dexcom app and xDrip.  That doesn't provide any useful info about distinction between xDrip and a variant xDrip.
Can you please show a screenshot of what you see on the lockscreen showing xDrip and variant so that I can see how they are not indistinguishable?  Then, your point will be clear.

**Reply by Pescador76** [reply]
![grafik](https://user-images.githubusercontent.com/13404652/209006580-2f5ea38f-8b21-4afb-8508-3e959b2a2062.png)

![grafik](https://user-images.githubusercontent.com/13404652/209006697-ff727952-e12b-441e-8837-a729022d9b7c.png)

**Comment by Pescador76** [comment]
In my example, I used the data from the same source because I have not yet made the switch to xdrip with nightscout for my second child. But I think the problem should be clear from this.

**Reply by Navid200** [reply]
This is what I see on my lock screen:
![P_20221221_164946_vHDR_Auto](https://user-images.githubusercontent.com/51497406/209009026-2bd628b2-d6f7-4cb7-9bec-eb1d36af1271.jpg)

The notifications on the lock screen are distinct to each app.

If you are using xDrip and a variant you see notifications that re not distinct to each app, I need to know how I can recreate the problem.

**Reply by Pescador76** [reply]
The problem, which is obvious from your screenshot as well as from mine, is that it is not known which notification belongs to which patient. It only says "xdrip+" or "xdrip+ variant1". I get confused every time I see this display and don't know exactly which patient which value belongs to. So please just add a label so that it says e.g. "xdrip+ [patient name 1]" and "xdrip+ variant1 [patient name 2]".

**Reply by Navid200** [reply]
Alright.  I am OK if you want to turn this into an issue.
Please note that is not a promise that this will be implemented.  But, since there have been multiple requests, we can have a feature request.  Bugs and feature requests are all issues.

So, if you like, please go ahead and convert this to an issue.  
Or, you can open a new issue.  If you do that, which will be fine also, please don't forget to include a link to this discussion in the first post right at the beginning.

**Comment by Navid200** [comment]
I wonder if you can perform a test.
The following talks about how you can rename an app:
https://9meters.com/technology/phones/how-to-rename-apps-on-android 

I haven't tried it.  Can you have a look to see if you rename the variants, would it also rename the widget on the lock screen?
